client/src/components/SearchBar/SearchBar.tsx

```diff
@@ -29,6 +29,26 @@ export function SearchBar({ onRoute, tripQueryVariables, setTripQueryVariables,
   const [showServerInfo, setShowServerInfo] = useState(false);
   const target = useRef(null);
 
+  const setFromLocation = useCallback(
```

Comment:
This is perfectly fine pattern, but we have done it slightly differently - passing tripQueryVariables and setTripQueryVariables into the input components, and calling them there. I suggest keeping to this for consistency.

Reply:
I moved these functions to the `LocationInputField` component.